message/router.go

```diff
@@ -430,6 +456,21 @@ func (h *handler) run(middlewares []HandlerMiddleware) {
 	h.logger.Debug("Router handler stopped", nil)
 }
 
+// AddMiddleware adds a new middleware to specified handler in the router.
+//
+// The order of middlewares matters. Middleware added at the beginning is executed first.
+func (h *handler) AddMiddleware(m ...HandlerMiddleware) {
+	h.logger.Debug("Adding middleware to handler", watermill.LogFields{
+		"count":       fmt.Sprintf("%d", len(m)),
+		"handlerName": h.name,
+	})
+
+	for _, handlerMiddleware := range m {
+		middleware := NewMiddleware(handlerMiddleware, h.name)
+		h.router.middlewares = append(h.router.middlewares, middleware)
```

Comment:
This looks like a violation of the Law of Demeter, it would be a bit more clean to introduce a method for this in router.